internal/querycoord/task_scheduler.go

```diff
@@ -801,9 +815,9 @@ func (scheduler *TaskScheduler) waitActivateTaskDone(wg *sync.WaitGroup, t task,
 
 		switch t.msgType() {
```

Comment:
seem here, why do we have a redo function for each task type,  but do a switch case here?